examples/basic/extended_scanner/main.go

```diff
@@ -28,8 +29,13 @@ func main() {
 	ble.SetDefaultDevice(d)
 	// ExtendedScan for specified durantion, or until interrupted by user.
 	fmt.Printf("ExtendedScaning for %s...\n", *du)
-	ctx := ble.WithSigHandler(context.WithTimeout(context.Background(), *du))
-	chkErr(ble.ExtendedScan(ctx, *dup, advHandler, nil))
+	for i := 0; i < 3; i++ {
+		ctx := ble.WithSigHandler(context.WithTimeout(context.Background(), *du))
+		err = ble.ExtendedScan(ctx, *dup, advHandler, nil)
+		if errors.Cause(err) == hci.ErrHCIHandlePacket {
+			chkErr(ble.Reset(ctx))
+		}
```

Comment:
Why does it do `ble.ExtendedScan()` three times?

Reply:
@zakkie 
Thanks!
I wanted to test whether the HCI could be Reset and then BLE scanned successfully when implemented in a deliberate way to enter the error branch.

https://github.com/moderepo/ble/pull/3/files/a1c6e82d413eac20ed1e093ebc1e85b084d58419#r1886082381